cmd/containerd-shim-runhcs-v1/task_hcs.go

```diff
@@ -596,6 +596,30 @@ func (ht *hcsTask) DumpGuestStacks(ctx context.Context) string {
 	return ""
 }
 
+func (ht *hcsTask) Share(ctx context.Context, req *shimdiag.ShareRequest) error {
+	if ht.host == nil {
+		return errors.New("task is not isolated")
+	}
+	// For hyper-v isolated WCOW the task used isn't the standard hcsTask so we
+	// only have to deal with the LCOW case here.
+	st, err := os.Stat(req.HostPath)
+	if err != nil {
+		return fmt.Errorf("could not open '%s' path on host: %s", req.HostPath, err)
+	}
+	var (
+		restrictAccess bool
+		fileName       string
+		allowedNames   []string
+	)
+	if !st.IsDir() {
+		_, fileName = filepath.Split(req.HostPath)
```

Comment:
Don't we also need to update the host path to just be the directory portion of the path?

Reply:
Ahhh shoot I think I screwed this up last night. Fixing